### PR TITLE
Refactor internals code

### LINKED ISSUE #18287: Space worthy helmets now provide telekinetic internals
### Severity

Exploit

### Short Description

Using a space worthy hemet for oxygen and dropping the oxygen tank then walking away keeps you breathing from that tank.

### Reproduction

Space worthy helmet
Turn internals on
Drop internals
Walk away
Internals continue

#17992 this PR caused it.

### Round ID

N/A

### Test Merges

_No response_

### BYOND Username

Namjay1170

### Was this done on a locally hosted or non-Yogstation hosted server?

None

### Additional information

#17992#17992#17992#17992#17992#17992#17992#17992#17992

### PR DESCRIPTION
**Pull request by ThatLing**
# Document the changes in your pull request

Reverts https://github.com/yogstation13/Yogstation/pull/17992 and ports https://github.com/tgstation/tgstation/pull/70750

Closes https://github.com/yogstation13/Yogstation/issues/18287

### CODE REVIEW

File: code/modules/mob/living/carbon/inventory.dm [hunk 1]
```diff
@@ -117,7 +117,161 @@
...
+ */
+/mob/living/carbon/proc/close_internals(is_external = FALSE)
+	var/obj/item/tank/target_tank = is_external ? external : internal
+	if (!target_tank)
+		return
+	if (is_external)
+		external = null
+	else
+		internal = null
+	target_tank.after_internals_closed(src)
```
Comment (Hopekz) [inline code/modules/mob/living/carbon/inventory.dm:198]:
```suggestion
	target_tank?.after_internals_closed(src)
```

File: code/modules/mob/living/carbon/inventory.dm [hunk 2]
```diff
@@ -117,7 +117,161 @@
...
+		return
+	to_chat(src, span_notice("You close [is_external ? external : internal] valve."))
+	return close_internals(is_external)
+
+/// Prepares emergency disconnect from open air tanks and notifies in chat. Usually called after mob suddenly unequips breathing apparatus.
+/mob/living/carbon/proc/cutoff_internals()
+	if (!external && !internal)
+		return
+	to_chat(src, span_notice("Your internals disconnect from [external || internal] and the valve closes."))
+	close_all_airtanks()
```
Comment (Hopekz) [inline code/modules/mob/living/carbon/inventory.dm:250]:
```suggestion
	close_all_airtanks()
	update_action_buttons_icon()
```

Comment (Hopekz) [reply code/modules/mob/living/carbon/inventory.dm:250]: I think you need this here to make sure the icon updates after you close them all.

File: code/modules/mob/living/carbon/inventory.dm (hunk 1 above)
Comment (ThatLing) [reply code/modules/mob/living/carbon/inventory.dm:198]: ``target_tank`` is already checked

File: code/modules/mob/living/carbon/inventory.dm (hunk 2 above)
Comment (ThatLing) [reply code/modules/mob/living/carbon/inventory.dm:250]: ``update_action_buttons_icon`` is called from ``close_all_airtanks``

**Review by Moltijoe — APPROVED** [review]

**Pull request merged**